synchronizer/synchronizer.go

```diff
@@ -827,6 +801,43 @@ func (s *ClientSynchronizer) ExecuteReorg(blockNumber uint64, reason string) err
 	return s.resetState(blockNumber - 1)
 }
 
+// checkReorgAndExecuteReset function will check if there is a reorg and execute the reset
+// returns true is reset have been done
+func (s *ClientSynchronizer) checkReorgAndExecuteReset(lastEthBlockSynced *state.Block) (bool, *state.Block, error) {
+	var err error
+	if lastEthBlockSynced == nil {
```

Comment:
this can't be nil. If this happens Fatal

Reply:
I'll change for a error